pageserver/src/tenant.rs

```diff
@@ -1350,13 +1268,64 @@ impl Tenant {
                         .map_err(LoadLocalTimelineError::ResumeDeletion);
                     }
 
-                    // We're loading fresh timeline that didnt yet make it into remote.
-                    (None, Some(remote_client))
+                    // see if there is at most one layer file and a metadata file in the directory
+                    // if not, we should fail the loading
+                    //
+                    // otherwise delete the not yet uploaded timeline.
+                    //
+                    // this timeline cannot have children because those are not allowed to be
+                    // created before uploading.
+
+                    let path = self.conf.timeline_path(&self.tenant_id, &timeline_id);
+
+                    return tokio::task::spawn_blocking({
```

Comment:
I think the comment would be a lot more understandable if the first sentence explained that we're dealing with not-yet-uploaded timelines here.

Then explain what the code inside  spawn_blocking does.
I.e., explain why it's better to "fail the loading" instead of just cleaning up and forgetting about this unfinished creation.

Reply:
592802a2041310340f120dde975ceae77f7d4497